Source/radiation/HypreABec.cpp

```diff
@@ -421,6 +421,162 @@ void HypreABec::hbcoef (const Box& bx,
     Gpu::synchronize();
 }
 
+void HypreABec::hbmat (const Box& bx,
+                       Array4<GpuArray<Real, AMREX_SPACEDIM+1>> const& mat,
+                       int cdir, int bct, Real bcl,
+                       Array4<int const> const& mask,
+                       Array4<Real const> const& b,
+                       Real beta, const Real* dx)
+{
+    bool xlo = false;
+    bool ylo = false;
+    bool zlo = false;
+
+    bool xhi = false;
+    bool yhi = false;
+    bool zhi = false;
+
+    Real h;
+
+    if (AMREX_SPACEDIM == 1) {
+
+        if (cdir == 0) {
+            xlo = true;
+            h = dx[0];
+        }
+        else if (cdir == 1) {
+            xhi = true;
+            h = dx[0];
+        }
+        else {
+            amrex::Error("Unknown cdir");
+        }
+
+    }
+    else if (AMREX_SPACEDIM == 2) {
+
+        if (cdir == 0) {
+            xlo = true;
+            h = dx[0];
```

Comment:
this is atleast the 3rd time we've had a PR with this logic in it.  Maybe we can extract this bit to an inline function?